### ISSUE TITLE
Long running merge relationship

### ISSUE THREAD

**Issue by AviAvni** [neo4j/neo4j#11962]
We want to upgrade our production to use version 3.4.X
We have problem running query like this in 3.4.2 but it's ok on 3.2.9
```
 MATCH (a:Profile {public_user_id: toUpper("700f55d3-d239-4c22-91b2-b102b78a7220")})
       MATCH (b:Profile {public_user_id: toUpper("e7fb691c-ca4d-4390-bea9-61c0eb1b1cac")})
       MERGE (a)-[:ProfileFavorites_2018_07_04]->(b);
0 rows available after 32 ms, consumed after another 0 ms
Created 1 relationships
```
but the query takes more then 2 minutes

when we are running it we are checking the locks and this is the result that we are getting
```
CALL apoc.monitor.locks(0);
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| advertedDeadLocks | lockCount | contendedLockCount | minimumWaitTimeMs | contendedLocks                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | info                                                                   |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| -1                | 6         | 6                  | 0                 | [{description: "ExclusiveLock[
Client[3] waits for []]", resourceId: "12935562", resourceType: "NODE"}, {description: "ExclusiveLock[
Client[3] waits for []]", resourceId: "12169038", resourceType: "NODE"}, {description: "SharedLock[
Client[3] waits for []]", resourceId: "-2027625510", resourceType: "INDEX_ENTRY"}, {description: "SharedLock[
Client[3] waits for []]", resourceId: "744812921", resourceType: "INDEX_ENTRY"}, {description: "SharedLock[
Client[3] waits for []]", resourceId: "0", resourceType: "LABEL"}, {description: "SharedLock[
Client[3] waits for []]", resourceId: "11504", resourceType: "RELATIONSHIP_TYPE"}] | "Showing contended locks where threads have waited for at least 0 ms." |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

**Comment by InverseFalcon** [comment]
Can you please supply the PROFILE plan of this query, with all elements of the plan expanded?

**Comment by NivTikolski** [comment]
here is the plan:
![plan](https://user-images.githubusercontent.com/7282159/42436842-f78b5c4a-8363-11e8-91fa-2d7c56835b06.png)

as avi said, the problem is when we ran the query, there are some locks(we are the only the connected to the graph), when the locks release, the query ran and it "looks" like it take 30ms, but we waited 2-3 mintues.
@InverseFalcon

**Comment by jexp** [comment]
Are there a lot of concurrent writes to these two profiles?

Are there other transactions that are not fully consumed or are still open/pending?

You can also list transactions in 3.4.x, please try to run `CALL dbms.listTransactions()` to see which other tx are ongoing

**Comment by AviAvni** [comment]
No transaction is running while running this transaction we have no traffic to this database

**Comment by noamket** [comment]
**@jexp is our savior!**
He has detected the cause of the issue,
It was our trigger that suffered from cypher regression in version 3.4.1 and made full db scans on each node update...

Please update this thread once the issue is resolved, we'll use @jexp workaround for now, 
Thanks **Michael**!

**Comment by sherfert** [comment]
The original regression in 3.4.1 has now been fixed. It will be included in 3.4.8
